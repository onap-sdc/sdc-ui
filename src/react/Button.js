import React from 'react';
import SVGIcon from './SVGIcon.js';

const Button = (props) => {
	let {btnType = 'default', color = 'primary', onClick, disabled, className, iconName, children, ...other} = props;
	return (
		<button onClick={onClick}
		        className={`sdc-button sdc-button-${btnType} sdc-button__${color} ${className || ''} ${iconName ? 'sdc-button__with-icon' : ''}`}
		        disabled={disabled} {...other}>
			<div className={iconName ? `sdc-button-link-icon` : null}>
				{iconName && <SVGIcon className='sdc-button-icon' name={iconName} />}
				<div className='sdc-button-label'>{children}</div>
			</div>
		</button>
	);
};

export default Button;
