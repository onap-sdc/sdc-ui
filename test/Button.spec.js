import React from 'react';
import Button from '../src/react/Button.js';

import HTMLButtonDefaultPrimary from '../components/button/button-default-primary.html';
import HTMLButtonDefaultWhite from '../components/button/button-default-white.html';
import HTMLButtonDefaultGray from '../components/button/button-default-gray.html';
import HTMLButtonDefaultPositive from '../components/button/button-default-positive.html';
import HTMLButtonDefaultNegative from '../components/button/button-default-negative.html';
import HTMLButtonDefaultWarning from '../components/button/button-default-warning.html';

import HTMLButtonOutlinePrimary from '../components/button/button-outline-primary.html';
import HTMLButtonOutlineGray from '../components/button/button-outline-gray.html';
import HTMLButtonOutlinePositive from '../components/button/button-outline-positive.html';
import HTMLButtonOutlineNegative from '../components/button/button-outline-negative.html';

import {mount} from 'enzyme';

describe('Button', () => {
  test('Button - Default - Primary', () => {
    const button = mount(<Button>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonDefaultPrimary);
  });

  test('Button - Default - White', () => {
    const button = mount(<Button color='white'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonDefaultWhite);
  });

  test('Button - Default - Gray', () => {
    const button = mount(<Button color='gray'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonDefaultGray);
  });

  test('Button - Default - Positive', () => {
    const button = mount(<Button color='positive'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonDefaultPositive);
  });

  test('Button - Default - Negative', () => {
    const button = mount(<Button color='white'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonDefaultNegative);
  });

  test('Button - Default - Warning', () => {
    const button = mount(<Button color='gray'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonDefaultWarning);
  });

  test('Button - Outline - Primary', () => {
    const button = mount(<Button type='outline'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonOutlinePrimary);
  });

  test('Button - Outline - Gray', () => {
    const button = mount(<Button type='outline' color='gray'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonOutlineGray);
  });

  test('Button - Outline - Positive', () => {
    const button = mount(<Button type='outline' color='positive'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonOutlinePositive);
  });

  test('Button - Outline - Negative', () => {
    const button = mount(<Button type='outline' color='negative'>Click Me</Button>);
    expect(button.html()).toEqual(HTMLButtonOutlineNegative);
  });
});
