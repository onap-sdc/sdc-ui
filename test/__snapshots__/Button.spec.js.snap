// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button - Default - Gray 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__gray  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Default - Negative 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__negative  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Default - Positive 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__positive  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Default - Primary - Disabled 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__primary  "
  disabled={true}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Default - Primary 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__primary  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Default - Warning 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__warning  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Default - White 1`] = `
<button
  className="sdc-button sdc-button-default sdc-button__white  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Link - Primary - Disabled 1`] = `
<button
  className="sdc-button sdc-button-link sdc-button__primary  "
  disabled={true}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Link - Primary - With Icon 1`] = `
<button
  className="sdc-button sdc-button-link sdc-button__primary  sdc-button__with-icon"
  disabled={undefined}
  onClick={undefined}
>
  <div
    className="sdc-button-link-icon"
  >
    <div
      className="svg-icon-wrapper  sdc-button-icon   bottom"
      disabled={undefined}
      onClick={undefined}
    >
      <svg />
      
    </div>
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Link - Primary 1`] = `
<button
  className="sdc-button sdc-button-link sdc-button__primary  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Outline - Gray 1`] = `
<button
  className="sdc-button sdc-button-outline sdc-button__gray  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Outline - Negative 1`] = `
<button
  className="sdc-button sdc-button-outline sdc-button__negative  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Outline - Positive 1`] = `
<button
  className="sdc-button sdc-button-outline sdc-button__positive  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;

exports[`Button Button - Outline - Primary 1`] = `
<button
  className="sdc-button sdc-button-outline sdc-button__primary  "
  disabled={undefined}
  onClick={undefined}
>
  <div
    className={null}
  >
    <div
      className="sdc-button-label"
    >
      Click Me
    </div>
  </div>
</button>
`;
